PDF.js vulnerable to arbitrary JavaScript execution upon opening a malicious PDF

##### Bug Report or Feature Request (mark with an `x`)
```
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [ x ] Bug report -> please search issues before submitting
- [ ] Feature request
- [ ] Documentation issue or request
```
Received a Dependabot vulnerability alert:

> PDF.js vulnerable to arbitrary JavaScript execution upon opening a malicious PDF
> Dependabot cannot update pdfjs-dist to a non-vulnerable version
> The latest possible version that can be installed is 2.16.105 because of the following conflicting dependencies:
> 
> ng2-pdf-viewer@10.0.0 requires pdfjs-dist@~2.16.105
> No patched version available for pdfjs-dist
> The earliest fixed version is 4.2.67.

I forked and pushed a [commit](https://github.com/VadimDez/ng2-pdf-viewer/commit/88e8f22d59565e4f9c8a91b440c5c764950b19f8), but there are build issues with pdfjs-dist. The [pdfjs github](https://github.com/mozilla/pdfjs-dist/blob/5f07d5a4159bb99eee2f6143d1297f03b45bba58/types/src/display/api.d.ts#L671C69-L672C5) repo does not contain any code related to `PromiseWithResolvers` or `WeakKey`.

```
./node_modules/pdfjs-dist/build/pdf.mjs - Error: Module parse failed: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/tools/babel/webpack-loader.js
 * ./node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
Error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)

./node_modules/pdfjs-dist/web/pdf_viewer.mjs - Error: Module parse failed: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/tools/babel/webpack-loader.js
 * ./node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
Error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)

Error: node_modules/pdfjs-dist/types/src/display/api.d.ts:686:23 - error TS2304: Cannot find name 'PromiseWithResolvers'.
686     _readyCapability: PromiseWithResolvers<any>;
                          ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/src/display/api.d.ts:738:18 - error TS2304: Cannot find name 'PromiseWithResolvers'.
738     _capability: PromiseWithResolvers<any>;
                     ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/src/display/api.d.ts:1381:23 - error TS2304: Cannot find name 'PromiseWithResolvers'.
1381     _readyCapability: PromiseWithResolvers<any>;
                           ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/src/display/text_layer.d.ts:101:18 - error TS2304: Cannot find name 'PromiseWithResolvers'.
101     _capability: PromiseWithResolvers<any>;
                     ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/web/pdf_find_controller.d.ts:105:27 - error TS2304: Cannot find name 'PromiseWithResolvers'.
105     _firstPageCapability: PromiseWithResolvers<any> | undefined;
                              ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/web/pdf_viewer.d.ts:296:27 - error TS2304: Cannot find name 'PromiseWithResolvers'.
296     _firstPageCapability: PromiseWithResolvers<any> | undefined;
                              ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/web/pdf_viewer.d.ts:297:33 - error TS2304: Cannot find name 'PromiseWithResolvers'.
297     _onePageRenderedCapability: PromiseWithResolvers<any> | undefined;
                                    ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/web/pdf_viewer.d.ts:298:23 - error TS2304: Cannot find name 'PromiseWithResolvers'.
298     _pagesCapability: PromiseWithResolvers<any> | undefined;
                          ~~~~~~~~~~~~~~~~~~~~

Error: node_modules/pdfjs-dist/types/web/text_layer_builder.d.ts:36:32 - error TS2304: Cannot find name 'WeakKey'.
36     textDivProperties: WeakMap<WeakKey, any>;
                                  ~~~~~~~
× Failed to compile.
```

Would you all know how to resolve the build issue?

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
   },
   "homepage": "https://github.com/VadimDez/ng2-pdf-viewer#readme",
   "dependencies": {
-    "pdfjs-dist": "^3.10.111",
+    "pdfjs-dist": "^3.11.174",
     "tslib": "^2.3.0"
   },
   "devDependencies": {

--- a/src/app/pdf-viewer/pdf-viewer.component.spec.ts
+++ b/src/app/pdf-viewer/pdf-viewer.component.spec.ts
@@ -114,7 +114,8 @@ describe('AppComponent', () => {
         url: src,
         cMapUrl,
         cMapPacked: true,
-        enableXfa: true
+        enableXfa: true,
+        isEvalSupported: false,
       });
     });
 
@@ -126,7 +127,8 @@ describe('AppComponent', () => {
         url: src,
         cMapUrl,
         cMapPacked: true,
-        enableXfa: true
+        enableXfa: true,
+        isEvalSupported: false,
       });
     });
 
@@ -139,7 +141,8 @@ describe('AppComponent', () => {
         url: srcUrl,
         cMapUrl,
         cMapPacked: true,
-        enableXfa: true
+        enableXfa: true,
+        isEvalSupported: false,
       });
     });
   });

--- a/src/app/pdf-viewer/pdf-viewer.component.ts
+++ b/src/app/pdf-viewer/pdf-viewer.component.ts
@@ -466,8 +466,9 @@ export class PdfViewerComponent
     const params: any = {
       cMapUrl: this._cMapsUrl,
       cMapPacked: true,
-      enableXfa: true
+      enableXfa: true,
     };
+    params.isEvalSupported = false; // http://cve.org/CVERecord?id=CVE-2024-4367
 
     if (srcType === 'string') {
       params.url = this.src;

